includes/class-sensei-course-participants.php

```diff
@@ -186,33 +208,46 @@ public function get_course_participant_count( $post_id = 0 ) {
 	}
 
 	/**
-	 * Get an array of learners taking the course
+	 * Get an array of learners taking the course.
 	 *
-	 * @since  1.0.0
-	 * @param  string $order    Order direction
-	 * @param  string $orderby  How to determine the order of learners
-	 * @return array  $learners The array of learners
+	 * @since 1.0.0
+	 * @since 2.0.1 Changed to get enrolled users instead of learners in progress.
+	 * 
+	 * @param  string $order    Order direction.
+	 * @param  string $orderby  How to determine the order of learners.
+	 *
+	 * @return array|false  $learners The array of learners.
+	 *                                If there are no users taking the course, returns `false`.
 	 */
 	public function get_course_learners( $order, $orderby ) {
-		$exclude_completed = $this->exclude_completed_participants( $this->get_course_id() );
-		$activity_args     = array(
-			'post_id' => absint( $this->get_course_id() ),
-			'type'    => 'sensei_course_status',
-			'number'  => 0,
-			'offset'  => 0,
-			'status'  => $exclude_completed ? 'in-progress' : 'any',
-		);
+		$post_id           = $this->get_course_id();
+		$exclude_completed = $this->exclude_completed_participants( $post_id );
 
-		$users = WooThemes_Sensei_Utils::sensei_check_for_activity( $activity_args, true );
+		if ( ! $this->use_legacy_enrolment_method() ) {
```

Comment:
I wonder if it makes more sense to have this logic inside `get_enrolled_participants_ids` and then have it call `get_enrolled_participants_ids_legacy` when `use_legacy_enrolment_method` is true? It might mean simplifying it here and in `get_course_participant_count`.

Reply:
I like your idea.
Is it like how you imagined? https://github.com/woocommerce/sensei-course-participants/pull/53/commits/df658598f4ca9978888a3f3eb5a904c7d0635102